### PR TITLE
Add two 2023 papers

### PR DESCRIPTION
**Pull request by jwallwork23**
We should also rebuild the website after merging this (I notice Jon's additions in #348 don't appear).

### CODE REVIEW

**Review by stephankramer — APPROVED** [review]
Thanks, yes we should rebuild indeed

Added a few more recent ones and fixed broken links in Jon's recent ones.

**Comment by jwallwork23** [comment]
All links look good. Have updated the webpage.

File: docs/source/publications.rst [hunk 1]
```diff
@@ -30,15 +30,20 @@ Hill, J., Rush, G., Peakall, J., Johnson, M., Hodson, L., Barlow, N.L.M.,
 Bowman, E.T., Gehrels, W.R., Hodgson, D.M., Kesserwani, G:
 Resolving tsunami wave dynamics: integrating sedimentology and numerical modelling.
 *The Depositional Record*
-doi: `10.1002/dep2.247 <https://doi.org/10.1002/dep2.247`>_, 2023.
+doi: `10.1002/dep2.247 <https://doi.org/10.1002/dep2.247>`_, 2023.
+
+Zhang, C., Zhang, J., Angeloudis, A., Zhou, Y., Kramer, S.C., Piggott, M.D.:
```
Comment (thangel) [inline docs/source/publications.rst:35]: I am not sure this paper used any thetis simulations... are we sure we want to list this here?

Comment (stephankramer) [reply docs/source/publications.rst:35]: Ah yes, you're quite right - that one shouldn't have been included

Comment (thangel) [reply docs/source/publications.rst:35]: I plan with @cpjordan to add some changes to some examples  shortly, maybe I can try to add a few more then and correct this

**Pull request merged**